client/src/main/java/io/pravega/client/admin/impl/ReaderGroupManagerImpl.java

```diff
@@ -95,23 +96,28 @@ public void createReaderGroup(String groupName, ReaderGroupConfig config) {
                                               new ReaderGroupStateUpdatesSerializer(), new ReaderGroupStateInitSerializer(), SynchronizerConfig.builder().build());
         Map<SegmentWithRange, Long> segments = ReaderGroupImpl.getSegmentsForStreams(controller, config);
         synchronizer.initialize(new ReaderGroupState.ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+        @Cleanup
+        ReaderGroupImpl groupImpl = (ReaderGroupImpl) getReaderGroup(groupName);
+        groupImpl.createState(config);
```

Comment:
though that was the original code, but perhaps lines 97 and 98 can also move inside createState method.

Reply:
That and we can probably remove lines 94-96 too